.github/workflows/release-preview.yml

```diff
@@ -0,0 +1,197 @@
+# Matomo release action for automated PREVIEW releases
+#
+# Required GitHub secrets:
+#
+# GPG_CERTIFICATE  |  ASCII armored or Base64 encoded GPG certificate that is used to create the signatures for the archives
+# GPG_CERTIFICATE_PASS  |  Passphrase of the GPG key
+
+name: Build preview release
+
+permissions:
+  actions: read # required for the tests job
+  checks: none
+  contents: write  # required to create tag and release
+  deployments: none
+  issues: read # required for the tests job
+  packages: none
+  pull-requests: read # required for the tests jobs
+  repository-projects: none
+  security-events: none
+  statuses: none
+
+on:
+  # TODO: remove manual dispatch after testing and enable cron
+  workflow_dispatch:
+    branches:
+      - 5.x-dev
+  #schedule:
+  #  - cron: '0 1 * * *' # 1am daily
+
+jobs:
+  prepare_preview_version:
+    runs-on: ubuntu-latest
+    outputs:
+      do_release: ${{ steps.changes.outputs.do_release }}
+      has_new_release: ${{ steps.version.outputs.has_new_version }}
+      new_version: ${{ steps.version.outputs.new_version }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          lfs: false
+          fetch-tags: true
+          fetch-depth: 0
+
+      - name: Prepare git config
+        run: |
+          cat <<- EOF > $HOME/.netrc
+            machine github.com
+            login $GITHUB_ACTOR
+            password $GITHUB_TOKEN
+            machine api.github.com
+            login $GITHUB_ACTOR
+            password $GITHUB_TOKEN
+          EOF
+          chmod 600 $HOME/.netrc
+          git config --global user.email "$GITHUB_ACTOR@users.noreply.github.com"
+          git config --global user.name "$GITHUB_ACTOR"
+          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Check if there are any changes to create a preview release for
+        id: changes
```

Comment:
That's a good suggestion, we can extract the check into a separate job, why not. Haven't done it that way because it was the first thing in the action when I started working on it and the tests and release jobs were added subsequently and at that point I haven't thought about also putting the check to a job.